src/ir/module-splitting.cpp

```diff
@@ -146,10 +146,21 @@ void TableSlotManager::addSlot(Name func, Slot slot) {
 }
 
 TableSlotManager::TableSlotManager(Module& module) : module(module) {
-  if (module.features.hasReferenceTypes()) {
+  Export* emscripten_table_export =
+    module.getExportOrNull("__indirect_function_table");
+  Table* indirect_table =
+    module.tables.size() == 1 ? module.tables[0].get() : nullptr;
+  bool emscripten_import = indirect_table->imported() &&
+                           indirect_table->module == "env" &&
+                           indirect_table->base == "__indirect_function_table";
+
+  if (module.features.hasReferenceTypes() && !emscripten_table_export &&
+      !emscripten_import) {
     // Just create a new table to manage all primary-to-secondary calls lazily.
     // Do not re-use slots for functions that will already be in existing
     // tables, since that is not correct in the face of table mutations.
+    // However, do not do this for emscripten; its loader code (and dynamic
+    // loading in particular) do not support this yet.
```

Comment:
Maybe move this part of the comment, or the entire block, up to the start of the function? In general having the comment before the code it explains seems best.